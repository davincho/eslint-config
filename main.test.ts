import path from 'node:path';
import url from 'node:url';

import spawn from 'cross-spawn';
import { expect, describe, it } from 'vitest';

const cwd = path.dirname(url.fileURLToPath(import.meta.url));

expect.addSnapshotSerializer({
  test: (val) => typeof val === 'string',
  print: (val: string) => {
    // Remove path information from snapshopts as this would make test fail on other machines
    return (val || '').replace(cwd, '.');
  }
});

const definitions = {
  'Incorrect JS file': {
    file: 'test-cases/incorrect.js'
  },
  'Correct JS file': {
    file: 'test-cases/correct.js'
  },
  'Incorrect TS file': {
    file: 'test-cases/incorrect.ts'
  },
  'Correct TS file': {
    file: 'test-cases/correct.ts'
  },
  'Incorrect TSX file': {
    file: 'test-cases/incorrect.tsx'
  },
  'Correct TSX file': {
    file: 'test-cases/correct.tsx'
  }
};

describe('Test files', () => {
  for (const [key, value] of Object.entries(definitions)) {
    it(key, () => {
      const { file } = value;

      const { output, stderr } = spawn.sync('./node_modules/.bin/eslint', [
        '--format',
        'tap',
        file
      ]);

      expect({
        outout: output.toString(),
        stderr: stderr.toString()
      }).toMatchSnapshot();
    });
  }
});
