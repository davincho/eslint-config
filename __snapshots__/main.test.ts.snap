// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test files Correct JS file 1`] = `
Object {
  outout: ,TAP version 13
1..1
ok 1 - ./test-cases/correct.js

,,
  stderr: ,
}
`;

exports[`Test files Correct TS file 1`] = `
Object {
  outout: ,TAP version 13
1..1
ok 1 - ./test-cases/correct.ts

,,
  stderr: ,
}
`;

exports[`Test files Correct TSX file 1`] = `
Object {
  outout: ,TAP version 13
1..1
ok 1 - ./test-cases/correct.tsx

,,
  stderr: ,
}
`;

exports[`Test files Incorrect JS file 1`] = `
Object {
  outout: ,TAP version 13
1..1
not ok 1 - ./test-cases/incorrect.js
  ---
  message: '''t'' is assigned a value but never used.'
  severity: error
  data:
    line: 2
    column: 7
    ruleId: '@typescript-eslint/no-unused-vars'
  messages:
    - message: Unexpected console statement.
      severity: error
      data:
        line: 5
        column: 1
        ruleId: no-console
  ...

,,
  stderr: ,
}
`;

exports[`Test files Incorrect TS file 1`] = `
Object {
  outout: ,TAP version 13
1..1
not ok 1 - ./test-cases/incorrect.ts
  ---
  message: An empty interface is equivalent to \`{}\`.
  severity: warning
  data:
    line: 2
    column: 11
    ruleId: '@typescript-eslint/no-empty-interface'
  messages:
    - message: '''Props'' is defined but never used.'
      severity: error
      data:
        line: 2
        column: 11
        ruleId: '@typescript-eslint/no-unused-vars'
  ...

,,
  stderr: ,
}
`;

exports[`Test files Incorrect TSX file 1`] = `
Object {
  outout: ,TAP version 13
1..1
not ok 1 - ./test-cases/incorrect.tsx
  ---
  message: There should be at least one empty line between import groups
  severity: error
  data:
    line: 1
    column: 1
    ruleId: import/order
  messages:
    - message: '''lodash'' is defined but never used.'
      severity: error
      data:
        line: 1
        column: 8
        ruleId: '@typescript-eslint/no-unused-vars'
    - message: '''testFunction'' is defined but never used.'
      severity: error
      data:
        line: 2
        column: 10
        ruleId: '@typescript-eslint/no-unused-vars'
    - message: '\`react\` import should occur before import of \`lodash\`'
      severity: error
      data:
        line: 5
        column: 1
        ruleId: import/order
    - message: '''TestComponent'' is assigned a value but never used.'
      severity: error
      data:
        line: 8
        column: 7
        ruleId: '@typescript-eslint/no-unused-vars'
  ...

,,
  stderr: ,
}
`;
